src/index.tsx

```diff
@@ -7,15 +7,19 @@ import App from "./components/App";
 import reportWebVitals from "./reportWebVitals";
 import theme from "theme/theme";
 import ErrorHandler from "components/error-handler/error-handler";
+import SyncHandler from "components/sync-handler/sync-handler";
 
 ReactDOM.render(
   <React.StrictMode>
     <Provider store={ store }>
       <ThemeProvider theme={ responsiveFontSizes(theme) }>
         <CssBaseline/>
-        <ErrorHandler>
-          <App/>
-        </ErrorHandler>
+        <SyncHandler>
+          <ErrorHandler>
```

Comment:
Should these be other way around? I assume that sync can also fail and throw error